File: doc-examples/hibernate-example-kotlin/src/main/kotlin/example/UserRepository.kt
package example

import io.micronaut.data.annotation.Fetch
import io.micronaut.data.annotation.Query
import io.micronaut.data.annotation.Repository
import io.micronaut.data.repository.CrudRepository
import jakarta.validation.constraints.NotNull
import java.util.stream.Stream

@Repository
interface UserRepository : CrudRepository<User, Long> {

    @Query("UPDATE User SET enabled = false WHERE id = :id")
    override fun deleteById(id: @NotNull Long)

    @Query("FROM User user WHERE enabled = false")
    fun findDisabled(): MutableList<User>


    // tag::fetch_size_streaming[]
    /**
     * Retrieves all users from the data store in batches.
     *
     * The returned Stream is lazily evaluated, meaning that it will only fetch data as it is consumed.
     * The fetch size is set to 1500, which controls the number of rows fetched from the database at a time.
     * It is the caller's responsibility to close the Stream to release any underlying resources.
     *
     * @return a Stream of all enabled users
     */
    @Fetch(1500)
    fun listAll(): Stream<User>

    /**
     * Retrieves all users from the data store.
     *
     * The returned Stream is lazily evaluated, meaning that it will only fetch data as it is consumed.
     * It is the caller's responsibility to close the Stream to release any underlying resources.
     *
     * @return a Stream of all users
     */
    fun queryAll(): Stream<User>

    // end::fetch_size_streaming[]
}

File: doc-examples/hibernate-example-python/src/main/python/example/UserRepository.py
from java.util.stream import Stream
from micronaut.data.annotation import Fetch
from micronaut.data.annotation import Query, Repository
from micronaut.data.repository import CrudRepository

from example.User import User


@Repository
class UserRepository(CrudRepository[User, int]):

    @Query("UPDATE Users SET enabled = false WHERE id = :id")
    def deleteById(self, id: int) -> None: ...

    @Query("FROM Users WHERE enabled = false")
    def findDisabled(self) -> list[User]: ...

    # tag::fetch_size_streaming[]
    @Fetch(1500)
    def listAll(self) -> Stream[User]:
        """Retrieves all users from the data store in batches.

        The returned Stream is lazily evaluated, meaning that it will only fetch data as it is consumed.
        The fetch size is set to 1500, which controls the number of rows fetched from the database at a time.
        It is the caller's responsibility to close the Stream to release any underlying resources.

        :return: a Stream of all enabled users
        """
        ...

    def queryAll(self) -> Stream[User]:
        """Retrieves all users from the data store.

        The returned Stream is lazily evaluated, meaning that it will only fetch data as it is consumed.
        It is the caller's responsibility to close the Stream to release any underlying resources.

        :return: a Stream of all users
        """
        ...
    # end::fetch_size_streaming[]
